src/screens/OtherProfile/index.js

```diff
@@ -414,20 +337,22 @@ const OtherProfile = () => {
             {__renderFollowerDetail()}
           </View>
         </View>
-        <BioAndChat
-          isAnonimity={isAnonimity}
-          bio={dataMain.bio}
-          openBio={openBio}
-          dataMain={dataMain}
-          isSignedMessageEnabled={isSignedMessageEnabled}
-          showSignedMessageDisableToast={showSignedMessageDisableToast}
-          loadingGenerateAnon={loadingGenerateAnon}
-          avatarUrl={profile.myProfile.profile_pic_path}
-          anonProfile={anonProfile}
-          username={dataMain.username}
-          toggleSwitch={toggleSwitch}
-          isAnonimityEnabled={isAnonimityEnabled}
-        />
+        {!isBlocked && !isBlocking && (
+          <BioAndChat
+            isAnonimity={isAnonimity}
+            bio={dataMain.bio}
+            openBio={openBio}
+            dataMain={dataMain}
+            isSignedMessageEnabled={isSignedMessageEnabled}
+            showSignedMessageDisableToast={showSignedMessageDisableToast}
+            loadingGenerateAnon={loadingGenerateAnon}
+            avatarUrl={profile.myProfile.profile_pic_path}
+            anonProfile={anonProfile}
+            username={dataMain.username}
+            toggleSwitch={toggleSwitch}
+            isAnonimityEnabled={isAnonimityEnabled}
+          />
+        )}
       </>
     );
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [311-354]

The conditions for rendering the `BioAndChat` component have been updated. Make sure that the new conditions (`!isBlocked && !isBlocking`) are equivalent to the old condition (`!blockStatus.blocked && !blockStatus.blocker`) in this context.



<!-- This is an auto-generated comment by OSS CodeRabbit -->